code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/github/GithubUtils.java

```diff
@@ -125,4 +148,58 @@ public String getTeamOrgById(String id) {
         Map<String, String> teamToOrg = (Map<String, String>) jsonData.get("teamToOrg");
         return  teamToOrg.get(id);
     }
+
+    public boolean isAllowed(List<String> idList, Set<ExternalId> externalIds){
+        boolean whiteListed = getWhitelistedUser(idList) != null;
```

Comment:
is getWhiteListedUser being called just from this method? If so - and we care only if the user is whitelisted or not - , would it make sense to change the return type of getWhiteListedUser to boolean? Its also a big unclear why the method with signature:

String getWhitelistedUser(List<String> idList

returns only one id from the idList passed in:

for (String id : idList) {
+            if (whitelist.contains(id)) {
+                return id;
+            }
+        }

I'm sure there is a logic behind it, but could you please elaborate it?